### PR TITLE
feat(internal-plugin-conversation): calculate fileHash on file upload

### PR DESCRIPTION
**Pull request by arpydi**
# Pull Request Template

## Description

These changes are a part of the new feature that is ANTI-MALWARE SCANNING. In share-activity.js in _upload function where we were making /finish API call, we are sending fileHash of the file that is uploaded in the body of the API call.

Note: Added a new package [crypto-js](http://npmjs.com/package/crypto-js) to generate file hash

Fixes # [SPARK-92936](https://jira-eng-gpk2.cisco.com/jira/browse/SPARK-92936)

## Type of Change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Test Coverage

Added a new test to cover fileHash sent in the body of /finish API call.
Test command: 

> npm test — —packages @webex/internal-plugin-conversation --node

Test Configuration:

SDK Version 1.80.29
Node/Browser Version 8.16.2
NPM Version 6.4.1

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by arpydi** [comment]
Hi @adamweeks 

This PR is an updated on of #1381
Since @meeshaT is on a long leave, I update the commit as per the guidelines and raised a new PR.
So, please consider this PR for File Hash Changes.

File: packages/node_modules/@webex/internal-plugin-conversation/test/unit/spec/share-activity.js [hunk 1]
```diff
@@ -74,5 +76,36 @@ describe('plugin-conversation', () => {
         assert.equal(sa._determineContentCategory(items), 'documents');
       });
     });
+    describe('#upload', () => {
+      let webex;
+      const fakeURL = `${process.env.ENCRYPTION_SERVICE_URL || 'https://encryption-a.wbx2.com'}/encryption/api/v1/keys/8a7d3d78-ce75-48aa-a943-2e8acf63fbc9`;
```
Comment (adamweeks) [inline packages/node_modules/@webex/internal-plugin-conversation/test/unit/spec/share-activity.js]: What is the purpose of using the env var in the unit test?

File: packages/node_modules/@webex/internal-plugin-conversation/test/unit/spec/share-activity.js [hunk 2]
```diff
@@ -74,5 +76,36 @@ describe('plugin-conversation', () => {
...
+        const spy = sinon.spy(webex.upload);
+        const inputData = {
+          phases: {
+            initialize: {
+              fileSize: 3333
+            },
+            finalize: {
+              body: {
+                fileSize: 3333,
+                fileHash: '4ea5c508a6566e76240543f8feb06fd457777be39549c4016436afda65d2330e'
```
Comment (adamweeks) [inline packages/node_modules/@webex/internal-plugin-conversation/test/unit/spec/share-activity.js]: Where does this file hash come from? Could we make it a variable and show where it originated from?

Comment (arpydi) [reply packages/node_modules/@webex/internal-plugin-conversation/test/unit/spec/share-activity.js]: The file hash comes from an encrypted fake URL.
We generate that hash through SHA 256 encryption function.
`const fileHash = sha256(file).toString;`

File: packages/node_modules/@webex/internal-plugin-conversation/test/unit/spec/share-activity.js (hunk 1 above)
Comment (arpydi) [reply packages/node_modules/@webex/internal-plugin-conversation/test/unit/spec/share-activity.js]: Removed the env var in this unit test.
@adamweeks please check it.

**Comment by adamweeks** [comment]
Thanks for making those changes. After another round there are a couple more changes requested:

* eslint is failing for your PR
* Please add the package.json changes as a separate commit with the message `feat(package): add crypto-js`
* Please squash the two plugin conversation commits.

**Comment by arpydi** [comment]
Hi @adamweeks 
In the process of squashing my commits, somehow I messed up my git workflow and accidentally pushed upstream/commits to this PR.
Due to that, I created another PR with all the changes mentioned above and I'm closing this PR.
Apologies for the disturbance created here.
New PR Link: https://github.com/webex/webex-js-sdk/pull/1437

**Pull request closed without merging**